tests/template/test_ring_template.py

```diff
@@ -46,3 +51,55 @@ def test_ring_default(hdf5_ds_2, tmpdir_factory, lt_ctx):
         results,
         expected['intensity'].raw_data,
     )
+
+
+@pytest.mark.dist
+@pytest.mark.asyncio
+def test_ring_tcp_cluster(lt_ctx):
+
+    conn = {"connection": {
+                    "type": "TCP",
+                    "address": "tcp://scheduler:8786"
+                    }
+            }
+    base_path = '/data/temp_data'
+    tmp_dir = os.path.join(base_path, 'tmp')
+    os.makedirs(tmp_dir)
+    ds_path = os.path.join(tmp_dir, 'tmp_random_hdf5.h5')
+    ds = create_random_hdf5(ds_path)
+    dataset = _get_hdf5_params(ds_path)
+
+    analysis = [{
+            "analysisType": "APPLY_RING_MASK",
+            "parameters": {
+                    'shape': 'ring',
+                    'cx': 8,
+                    'cy': 8,
+                    'ri': 5,
+                    'ro': 8,
+                    }
+            }]
+
+    notebook = notebook_generator(conn, dataset, analysis, save=True)
+    notebook = io.StringIO(notebook.getvalue())
+    nb = nbformat.read(notebook, as_version=4)
+    ep = ExecutePreprocessor(timeout=600)
+    out = ep.preprocess(nb, {"metadata": {"path": tmp_dir}})
+    data_path = os.path.join(tmp_dir, 'ring_result.npy')
+    results = np.load(data_path)
+
+    analysis = lt_ctx.create_ring_analysis(
+                            dataset=ds,
+                            cx=8,
+                            cy=8,
+                            ri=5,
+                            ro=8
+                        )
+    expected = lt_ctx.run(analysis)
+
+    assert np.allclose(
+        results,
+        expected['intensity'].raw_data,
+    )
+    # cleanup files after test
+    shutil.rmtree(tmp_dir, ignore_errors=True)
```

Comment:
This cleanup won't run if the test fails for some reason - it may be a good idea to create a fixture for this (it can even be local to the test file), something like this:


```
@pytest.fixture
def some_name():
    os.makedirs(...)
    ds = ... # create your data
    yield ds
    shutil.rmtree(tmp_dir, ignore_errors=True)
```

The same can be done by wrapping the whole test case into a `try...finally`, but that can make the code a bit harder to read.